### DISCUSSION TITLE
plugin was built with a different version of package internal/unsafeheader

### DISCUSSION THREAD

**Discussion by Tanz0rz** [golangci/golangci-lint#2529] [Q&A]
Hello!

I am trying to setup a private linter with golangci-lint and am following the docs on [this page](https://golangci-lint.run/contributing/new-linters/#how-to-add-a-private-linter-to-golangci-lint) to do it. 

I am seeing this error when I try to run my private linter as part of `golangci-lint run`:

`ERRO Unable to load custom analyzer example:./example.so, plugin.Open("/Users/me/project/example"): plugin was built with a different version of package internal/unsafeheader`

My linter is quite simple and has a single dependency in its go.mod file:

```
module github.com/Tanz0rz/go-printf-func-name

go 1.16

require (
	golang.org/x/tools v0.1.9-0.20211228192929-ee1ca4ffc4da
)
```

I also went ahead and built the plugin with the latest go 1.16 version to try my best to match the data I pulled from the go.mod file in this repo. 

I am not sure what I am doing wrong.

**Comment by Tanz0rz** [comment]
Any suggestions here?

**Comment by behzadsh** [comment]
I had the same issue with a different package. I get `plugin was built with a different version of package golang.org/x/sys/execabs`.
did you find any solution?

**Reply by Tanz0rz** [reply]
Unfortunately, no

**Comment by ldez** [comment]
Hi,

plugins and golangci-lint must be compiled with the exact same stack, it's related to how Go plugins work.

It's a constraint that is not possible to overcome because it's a Go constraint.

**Reply by hermanbanken** [reply]
We could make it very easy for users to do this by exposing a workflow to make that possible, right? Like ~having golangci-lint emit the git sha it used to build~ (it does that) & use the go.mod file to output `go get lib@version` commands to get the same versions.

```
golangci-lint --version
golangci-lint has version 1.50.0 built from 704109c on 2022-10-02T11:12:24Z
```

My workflow to get it working was:

1. get sha from `golangci-lint --version`
2. inspect GitHub Action pipeline for build to see Go version used
3. use [https://go.dev/doc/manage-install#installing-multiple](https://go.dev/doc/manage-install#installing-multiple) to install the right go version
4. build using the right go version
5. inspect any remaining package issues and match versions with https://github.com/golangci/golangci-lint/blob/704109c/go.mod

This worked fine, but is super brittle and will not fly in CI/CD if golangci-lint isn't installed using an exact version; and probably instantly and horribly breaks once golangci-lint is updated.
